### ISSUE TITLE
Ability to define and reuse schemas for primative types

### ISSUE THREAD

**Issue by elee1766** [fabien0102/ts-to-zod#269]
## Feature description

I would like to define a schema for an existing type somehow, and use that instead of defining a format everywhere.  This could potentially also allow third party package support via type aliases.

For instance, it would be nice to be able to do something like
```ts
/**
 *  0x prefixed hexadecimal string
 *
 *  @pattern ^0x[a-fA-F0-9]*$
 */
type Hex = `0x${string}`

export interface Foo {
   lorum: Hex
}
export interface Bar {
   ipsum: Hex
}
```

which would ideally output something like

```ts
export const hexSchema = z.string().regex(/^0x[a-fA-F0-9]*$/)
export const fooSchema = z.object({lorum: hexSchema})
export const barSchema = z.object({ipsum: hexSchema})
```

I have not looked too deeply into the code - does something like this seem feasible?

**Comment by tvillaren** [comment]
Hello @elee1766 

I'm not sure what you're asking 😄 

👉 `ts-to-zod` does not support the template literal `0x${string}` but the following input: 
```ts
/**
 *  0x prefixed hexadecimal string
 *
 *  @pattern ^0x[a-fA-F0-9]*$
 */
export type Hex = string;

export interface Foo {
  lorum: Hex;
}
export interface Bar {
  ipsum: Hex;
}
```

will output:
```ts
// Generated by ts-to-zod
import { z } from "zod";

export const hexSchema = z.string().regex(/^0x[a-fA-F0-9]*$/);

export const fooSchema = z.object({
  lorum: hexSchema,
});

export const barSchema = z.object({
  ipsum: hexSchema,
});
```

👉 it also supports imports, so if you you want to have 
```ts
/**
 *  @pattern ^0x[a-fA-F0-9]*$
 */
export type Hex = string;
```
in a specific file — say `hexa.ts` —  and the corresponding generated zod validator in `hexa.zod.ts` then, if you them in your `ts-to-zod.config.js` like that: 
```js
/**
 * ts-to-zod configuration.
 *
 * @type {import("./src/config").TsToZodConfig}
 */
module.exports = [
  { name: "hexa", input: "src/hexa.ts", output: "src/hexa.zod.ts" },
];
```

any file importing `Hex` from `hexa.ts` will automatically import the validator in its generated zod file: 
```ts
// Input

import { Hex } from "./hexa";
export interface Foo {
  lorum: Hex;
}

// Output

import { z } from "zod";
import { hexSchema } from "./hexa.zod";

export const fooSchema = z.object({
  lore: hexSchema
});
```

👉 As for 3rd party libraries, if you have a specific example, we could evolve the Config specification to enable imports of Zod Schema from those.

**Comment by elee1766** [comment]
oh, maybe it was the templated literal that was messing with things then? It wasn't working for me. Let me try again and see what happens.

thanks!

**Comment by tvillaren** [comment]
Hey @elee1766, 
Have you had a chance to try it again?

**Comment by elee1766** [comment]
hey yeah!

so it's working for us for string types but not string template types.

it would be nice if we could also get it working for a string template type. is there any way we can hack around it ?

**Comment by tvillaren** [comment]
Hey, 
I don't think it is very easy to handle this kind of string template with JSDoc tag, there would be too many specific cases to handle IMO. 

A workaround you can rely on is the use of the `@schema` tag: 

```ts

// input
/**
 *  0x prefixed hexadecimal string
 *
 *  @schema string().regex(/^0x[a-fA-F0-9]*$/)
 */
export type Hex = `0x${string}`;

// output
// Generated by ts-to-zod
import { z } from "zod";

export const hexSchema = z.string().regex(/^0x[a-fA-F0-9]*$/);
```

You basically can define whatever you want to be output after the `z` for your schema that way.

**Comment by elee1766** [comment]
that is good enough i think!
